sourcegraph/sourcegraph
# Insights/shareable-link-modal

Closes https://github.com/sourcegraph/sourcegraph/issues/34766

## Test plan

- Make sure that you can see the "share insight modal UI" by clicking on the context menu item - get a sharable link
- Make sure that the content of this UI is different based on a subset of dashboards in which this insight is included (if it's included in any global dashboard, you see that anyone can access this dashboard. If it's included in any org dashboard, you see a list of organizations that users should be part of to be able to access this insight by share link, and if it's is included in personal dashboards or in no dashboards, you should see that this insight is private and only you can access it by sharable link) See the original issue for more details about this states https://github.com/sourcegraph/sourcegraph/issues/34766
- Make sure that you copy that link and see the standalone insight page by this link

## App preview:

- [Web](https://sg-web-insights-shareable-link-modal.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gafooqqxcd.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
